app/core/pokemon-entity.ts

```diff
@@ -122,7 +122,7 @@ export class PokemonEntity extends Schema implements IPokemonEntity {
   isClone = false
   refToBoardPokemon: IPokemon
   commands = new Array<SimulationCommand>()
-  effectsSet = new Set<EffectClass>()
+  effectsSet = new Set<DynamicEffect>()
```

Comment:
not sure what you call DynamicEffect. All effects are dynamic, since they are triggered on a certain condition ?